### DISCUSSION TITLE
Linux or Windows compiling?

### DISCUSSION THREAD

**Discussion by bakaneko59** [Jyers/Marlin#601] [Q&A] [answered]
All things considered, is it easier/better to build this firmware in a Windows or a Linux environment? I'm primarily a windows user but can easily build a Linux virtual machine on my server to work with if a Linux environment is more straightforward. I'm a complete newb when it comes to using GIT and compiling anything from source code.

Anyone have a link that provides good step by step instructions on the build procedure?

**Comment by tome9111991** [comment, marked as answer]
I complie with visual code and PlatfrimIO

Visual code is available for windows and linux

**Reply by aryajuanda** [reply]
I use visual studio code with platformio and marlin autobuild on mac os..
